Admin/components/AdminSite/Profile.php

```diff
@@ -103,6 +109,51 @@ protected function validate()
 				);
 			}
 		}
+
+		if ($this->ui->getWidget('two_fa_token')->value !== null) {
+			$this->validate2Fa();
+		}
+	}
+
+	// }}}
+	// {{{ protected function validate2Fa()
+
+	protected function validate2Fa()
+	{
+		// strip all non numeric characters like spaces and dashes that people
+		// might enter (e.g. Authy adds spaces for readability)
+		$token = preg_replace(
+			'/[^0-9]/',
+			'',
+			$this->ui->getWidget('two_fa_token')->value
+		);
+
+		// The timestamp is used to make sure this, or tokens before this,
+		// can't be used to authenticate again. There's a "window" of token
+		// use and without this, someone could capture the code, and re-use it.
+		$two_fa = new TwoFactorAuth();
+		$success = $two_fa->verifyCode(
+			$this->app->session->user->two_fa_secret,
```

Comment:
The profile page uses `$this->data_object` to refer to the account in most places. Does it make sense here too?